### ISSUE TITLE
Certificate validation failed

### ISSUE THREAD

**Issue by ktry1** [omnia-network/ic-websocket-gateway#37]
### good day, I am using ic-websocket v1.4.1 with frontend and getting error on open:

[certification] Witness != Tree passed in ic-certification utils.ts:55:11
    isMessageBodyValid utils.ts:55
    _isIncomingMessageValid ic-websocket.ts:510
    _processIncomingMessage ic-websocket.ts:326
    _processNext queues.ts:84
    process queues.ts:69
    addAndProcess queues.ts:51
    _onWsMessage ic-websocket.ts:233

### at the same time another error is printed in console:
[onWsMessage] Error: [onWsMessage] Certificate validation failed
    _processIncomingMessage ic-websocket.ts:328
    _processNext queues.ts:84
    process queues.ts:69
    addAndProcess queues.ts:51
    _onWsMessage ic-websocket.ts:233
ic-websocket.ts:349:13
    _processIncomingMessage ic-websocket.ts:349
    _processNext queues.ts:84
    process queues.ts:69
    addAndProcess queues.ts:51
    _onWsMessage ic-websocket.ts:233

### on error callback gives me this:
Error: 
error { target: null, isTrusted: false, message: "", filename: "", lineno: 0, colno: 0, error: Error, eventPhase: 0, bubbles: false, cancelable: false, … }
​
bubbles: false
​
cancelBubble: false
​
cancelable: false
​
colno: 0
​
composed: false
​
currentTarget: null
​
defaultPrevented: false
​
error: Error: Error receiving message: Error: [onWsMessage] Certificate validation failed
​
eventPhase: 0
​
explicitOriginalTarget: null
​
filename: ""
​
isTrusted: false
​
lineno: 0
​
message: ""
​
originalTarget: null
​
returnValue: true
​
srcElement: null
​
target: null
​
timeStamp: 6972
​
type: "error"
​
<get isTrusted()>: function isTrusted()
​
<prototype>: ErrorEventPrototype { message: Getter, filename: Getter, lineno: Getter, … }
Lobby.tsx:97:24

### My code at frontend:

const icUrl = process.env.DFX_NETWORK === "local" ? "http://127.0.0.1:4943" : "https://ic0.app";
const gatewayUrl = process.env.DFX_NETWORK === "local" ? "ws://127.0.0.1:8080" : "wss://gateway.icws.io";
const wsConfig = createWsConfig({
canisterId: process.env.CANISTER_ID_GATEWAY!,
canisterActor: web3State.Gateway_actor,
identity: web3State.identity,
networkUrl: icUrl,
});
let websocket = new IcWebSocket(gatewayUrl, undefined, wsConfig);

### Could you please help me pinpoint what is wrong?

**Comment by ilbertt** [comment]
> Env variables are loaded correctly, I checked it in console.

Does this mean that `DFX_NETWORK` = `local` and that `CANISTER_ID_GATEWAY` = the canister id of the Motoko canister on which the [IC WS Motoko CDK](https://github.com/omnia-network/ic-websocket-cdk-mo) is installed and set up?

It looks like the certified data are **not** looked up correctly from the local replica certificate, and the causes can be many.

Also, have you tried resetting/reinstalling the Motoko canister before opening the frontend?

**Comment by ktry1** [comment]
#### Good day, thank you for your reply! Here is the Gateway.mo canister that I am using: https://github.com/ktry1/Gateway_test/blob/main/Gateway.mo. 
#### I have restarted pc, and deployed all canisters anew multiple times.
#### Variables that I am passing into createWsConfig, as gotten from console.log:

##### process.env.DFX_NETWORK: "local"
##### process.env.CANISTER_ID_GATEWAY: "be2us-64aaa-aaaaa-qaabq-cai"
##### canisterActor: 
" Object { send_message_to_clients: handler(args), ws_close: handler(args), ws_get_messages: handler(args), ws_message: handler(args), ws_open: handler(args), Symbol("ic-agent-metadata"): {…} }
​
send_message_to_clients: function handler(args)
​
ws_close: function handler(args)
​
ws_get_messages: function handler(args)
​
ws_message: function handler(args)
​
ws_open: function handler(args)
​
Symbol(ic-agent-metadata): Object { config: {…}, service: {…} }
​
<prototype>: Object { … }
Lobby.tsx:83:20
"
##### identity: 
"Object { _inner: {…}, _delegation: {…}, _principal: {…} }
​
_delegation: Object { delegations: (1) […], publicKey: ArrayBuffer }
​
_inner: Object { _keyPair: {…}, _derKey: ArrayBuffer, _subtleCrypto: SubtleCrypto }
​
_principal: Object { _arr: Uint8Array(29), _isPrincipal: true }
​
<prototype>: Object { … }
Lobby.tsx:84:20
"
##### networkUrl: "http://127.0.0.1:4943"

**Comment by ilbertt** [comment]
I think I found out what's causing this error. The problem is with the `@dfinity/...` dependecies. Due to https://github.com/dfinity/agent-js/pull/878, the certificates are handled a bit differently starting from [v1.4.0](https://github.com/dfinity/agent-js/releases/tag/v1.4.0).

My suggestion is to manually install the `@dfinity/...` packages dependencies with version **v1.3.0** manually. Steps (execute them inside the frontend canister folder, if you've bootstrapped the project using the `dfx new` command):

1. remove the `ic-websocket-js` package:
    ```bash
    npm remove ic-websocket-js
    ```
2. remove the `@dfinity/...` packages:
    ```bash
    npm remove @dfinity/agent @dfinity/candid @dfinity/identity-secp256k1 @dfinity/principal
    ```
3. install the `@dfinity/...` packages with the correct version:
    ```bash
    npm install @dfinity/agent@1.3.0 @dfinity/candid@1.3.0 @dfinity/identity-secp256k1@1.3.0 @dfinity/principal@1.3.0
    ```
4. install the `ic-websocket-js` package:
    ```bash
    npm install ic-websocket-js
    ```
5. build and deploy the canisters again:
    ```bash
    dfx deploy
    ```

I've tested these steps locally, and it seems that the issue is solved.

We'll update the SDKs packages to use the newest dependencies soon, so you won't need this workaround anymore.

I'm sorry for the late reply, feel free to close this issue if this is the solution.
